conda_smithy/mamba_wrapper.py

```diff
@@ -0,0 +1,84 @@
+import json
+import logging
+
+logger = logging.getLogger(__name__)
+
+
+class ResolveFallback:
+    """
+    Wrapper that implements a tiny subset of :class:`conda_build.conda_interface.Resolve`.
+
+    Calls are first tried using mamba and we switch to conda_build.conda_interface on failure.
+    """
+
+    def __init__(self):
+        self.use_fallback = False
+        self.conda_resolve = None
+        self.mamba_pool = None
+        self.mamba_repos = None
+
+    def _load_mamba_repos(self):
+        import mamba.mamba_api as api
+        from mamba.utils import get_index, init_api_context
+
+        if self.mamba_repos is None:
+            if self.mamba_pool is None:
+                init_api_context()
+                self.mamba_pool = api.Pool()
+
+            index = get_index(("conda-forge",), prepend=False)
+            self.mamba_repos = []
+
+            for subdir, channel in index:
+                if subdir.loaded() == False and channel.platform != "noarch":
+                    # ignore non-loaded subdir if channel is != noarch
+                    continue
+
+                repo = api.Repo(
```

Comment:
everything looks great! 

This is just a heads-up: we are about to change the API slightly (will release a 0.3.0 soon-ish) where one should call `subdir.create_repo(pool)` instead of this.
So maybe pin mamba to 0.2.* for now?

Reply:
We can add run_constrained to the recipe.